### DISCUSSION TITLE
Passing `python_files` option to `pytest.Config`

### DISCUSSION THREAD

**Discussion by ktbarrett** [pytest-dev/pytest#10052] [Q&A] [answered]
I support a Python-based HDL testing framework [cocotb](https://github.com/cocotb/cocotb) where we use pytest's assertion rewriting [like so](https://github.com/cocotb/cocotb/blob/ffff390b600f5a64885528fd1461cab64d6657fd/cocotb/__init__.py#L322-L328).

```python
# Install the assertion rewriting hook, which must be done before we
# import the test modules.
from _pytest.assertion import install_importhook
from _pytest.config import Config

pytest_conf = Config.fromdictargs({}, ["--capture=no"])
install_importhook(pytest_conf)
````

However, this only applies assertion rewriting to modules that match `test_*.py`, which many users find a little annoying seeing as our current system has the user specify the test modules manually. The `python_files` config option seems to do what I want (tested by putting it in a `pytest.ini` file), but passing it to the `Config` constructor doesn't seem to work. The test for this option all write out an ini file so is `python_files` somehow a "config file only" option, or am I missing something?

```python
# What I tried
Config.fromdictargs({"python_files": "*.py"}, ["--capture=no"])
```

**Comment by nicoddemus** [comment, marked as answer]
Hi @ktbarrett sorry for the delay.

Any config option can be passed via command-line using `-o`, for example `pytest -o python_files=*.py`.

See the starting paragraph of [Configuration Options](https://docs.pytest.org/en/stable/reference/reference.html?highlight=command%20line#configuration-options).

Hope that helps!

**Reply by ktbarrett** [reply]
I tried the following based on your response and assertion rewriting occured, thanks! 

```python
pytest_conf = Config.fromdictargs({}, ["--capture=no", "-o", "python_files=*.py"])
install_importhook(pytest_conf)
```

As a matter of curiosity, what's the intended way to use the `option_dict` argument in `Config.fromdictargs`?

**Reply by RonnyPfannschmidt** [reply]
its a "broken" hack thats there for pytest-xdist - unfortunately its a larger refactoring to fix it

bascially its supposed to be possible to pass python-files inside of that, but its not working

**Reply by ktbarrett** [reply]
One more follow on question. How do you specify multiple python_files on the command line? I tried to follow the documentation's example of separating the elements with a space like `Config.fromdictargs({}, ["--capture=no", "-o", "python_files='a.py b.py'"])`, but that does not seem to work. Neither does separating the elements with commas.

EDIT: nvm, it seemed not to like using the quotation marks when there were multiple elements (`'a.py b.py'`), but was fine with them with one element (`'a.py'`) which led me astray. Space separated like so works (`python_files=a.py b.py`).
